examples/simple-example/README.md

```diff
@@ -35,3 +35,67 @@ miranum-core. The project consists of three modules:*
 Now you can simply start the process instance and view the log in the application console.
 
 <p align="right">(<a href="#top">back to top</a>)</p>
+
+### Generate element templates
+
+This examples also demonstrates how you can make use of the miranum element-templates generation feature
+which lets you generate element templates for your worker definitions. \
+To use the element template generation feature you have to add the @GenerateElementTemplates annotation to your worker definition method.
+
+To be able to use this annotation you have to add the miranum-core dependency or just the element-template-api dependency to your maven project.
+
+```xml
+<dependency>
+   <groupId>io.miragon.miranum</groupId>
+   <artifactId>element-template-api</artifactId>
+   <version>0.1.0-SNAPSHOT</version>
+</dependency>
+```
+
+When using the @GenerateElementTemplate annotation you have to specify some properties of your element template.
+
+```java
+@Worker(type = "sendMessage")
+@GenerateElementTemplate(name = "Send Message",
+        id = "send-message",
+        type = "sendMessage",
+        appliesTo = {BPMNElementType.BPMN_SERVICE_TASK},
+        version = "0.0.1")
+public Answer sendMessage(final SendMessageCommand message) {
+        return new Answer("answer to: " + message.getName());
+}
+```
+
+Additionally, when deciding for which Workflow-Engine you are going to build your project for you have to add the concrete
+dependency of the element-templates generator to your maven project. \
+Here for Camunda 7:
+
+```xml
+<dependency>
+   <groupId>io.miragon.miranum</groupId>
+   <artifactId>element-templates-c7</artifactId>
+   <version>0.1.0-SNAPSHOT</version>
+</dependency>
+```
+
+To further customize the generated element templates you can add the @ElementTemplateProperty annotation to the input and output parameter of your worker definition. \
+For example we can change the name, type and constraints on the properties of the SendMessageCommand:
+
+```java
+class SendMessageCommand {
+    @ElementTemplateProperty(name = "Message", type = "String", required = true)
+    private String name;
+
+    @ElementTemplateProperty(name = "Correlation Key", type = "String", required = true)
+    private String key;
+}
+```
+
+The element templates are then generated on startup of the main project. \
+Currently they are stored inside the `element-templates` directory in the resource folder of the main project.
```

Comment:
💭 Remember to update this sentence in case the generated files are stored somewhere else.